### PR TITLE
Add Sample Testplans to Documentation

### PR DESCRIPTION
**Pull request by vijaybabu4589**
# Summary

@yahoo/ychaos-dev 

**Fixes**: https://github.com/yahoo/ychaos/issues/31

#### Prerequisites
- [x] I have read the contribution guidelines

#### Code Analysis
- [ ] Covered by Unittests
- [ ] Security warnings ignored (Why?!)

#### Project related
- [ ] Schema changed and is backward compatible.
- [ ] Introduces a new sub-command for ychaos CLI

### Autogenerated Files
- [ ] Auto generated schema

### Pre-Merge Checklist

- [x] All the build jobs are passing

---

### Checklist (Reviewer 1)

- [ ] Reviewed Source Code
- [ ] Reviewed Tests (If applicable)
- [ ] Reviewed Documentation (If applicable)

### Checklist (Reviewer 2)

- [ ] Reviewed Source Code
- [ ] Reviewed Tests (If applicable)
- [ ] Reviewed Documentation (If applicable)

### CODE REVIEW

File: docs/testplan/sample_test_plan.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+```yaml
+...
+agents:
+  - type: traffic_block
+    config:
+      duration: 60
+      start_delay: 0
+      hosts:
+        - 'www.yahoo.com'
+        - 'www.google.com'
```
Comment (shashankrnr32) [inline docs/testplan/sample_test_plan.md:40]: Lets use some real time example of APIs for these like AWS/Redis APIs

File: docs/testplan/sample_test_plan.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+...
+agents:
+  - type: traffic_block
+    config:
+      start_delay: 0
+      duration: 60
+      hosts:
+        - "yahoo.com"
+        - "google.com"
+```
```
Comment (shashankrnr32) [inline docs/testplan/sample_test_plan.md:96]: Repeated?

File: docs/testplan/sample_test_plan.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+#### Endpoint Certificate Validation
+```yaml
+...
+agents:
+  - type: urls
+    config:
+      expiry_threshold: 7 # check if cert expire date is more that 7days
+      urls:
+        - "www.yahoo.com"
+        - "www.google.com"
```
Comment (shashankrnr32) [inline docs/testplan/sample_test_plan.md:107]: Lets use some URLs from Yahoo only (www.aol.com, www.techcrunch.com)

File: docs/testplan/sample_test_plan.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  - delay_before: 10
+  - state: 'STEADY'
+  - type: 'http_request'
+  - config:
+      latency: 100
+      urls:
+        - 'api.nationalize.io'
+      method: 'GET'
+      params:
+        name: 'india'
```
Comment (shashankrnr32) [inline docs/testplan/sample_test_plan.md:125]: Just want to know is there an API like this? No need to change but just for information

File: docs/testplan/sample_test_plan.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+...
+verification:
+  - delay_before: 10
+  - state: 'STEADY'
+  - type: 'sdv4'
+  - config:
+      sd_api_url: 'api.screwdriver.cd'
+      pipeline_id: 123123 # SD pipeline id
+      job_name: 'Run_FT' # SD job name
+      sd_api_token: 'API token' # Refer Screwdriver Docs
```
Comment (shashankrnr32) [inline docs/testplan/sample_test_plan.md:146]: Lets use the environment variable type of secret here.
```suggestion
      sd_api_token: # Refer Screwdriver Docs
          type : env
          id: SD_API_TOKEN
```

File: docs/testplan/sample_test_plan.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
+#### CPU Burn
```
Comment (shashankrnr32) [inline docs/testplan/sample_test_plan.md:1]: Add a header note 
```
This section of the document contains some example testplans for you to get started quickly on the project
```

File: docs/testplan/sample_test_plan.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+      sd_api_token: 'API token' # Refer Screwdriver Docs
+attack:
+    target_type: machine
+    agents:
+        - type: no_op
+```
+!!! note ""
+    For more information on verification plugins [refer](/ychaos/verification/)
+
+
```
Comment (shashankrnr32) [inline docs/testplan/sample_test_plan.md:155]: nit: Extra Line

File: docs/testplan/sample_test_plan.md (hunk 6 above)
Comment (shashankrnr32) [reply docs/testplan/sample_test_plan.md:1]: And how about renaming the file to `examples.md` and the page title to "Test Plan Examples"

**Review by shashankrnr32 — COMMENTED** [review]
Minor comments. Looks good overall

File: docs/testplan/sample_test_plan.md (hunk 4 above)
Comment (vijaybabu4589) [reply docs/testplan/sample_test_plan.md:125]: yes there is a API `https://api.nationalize.io/?name=india`

File: docs/testplan/sample_test_plan.md (hunk 2 above)
Comment (vijaybabu4589) [reply docs/testplan/sample_test_plan.md]: yes.. removed

File: docs/testplan/sample_test_plan.md [hunk 8]
```diff
@@ -0,0 +1,153 @@
+This section of the document contains some example testplans for you to get started quickly on the project
+#### CPU Burn
+```yaml
+description: Increase CPU utlization on all the cores for 60 second
```
Comment (r-r-2) [inline docs/testplan/sample_test_plan.md:4]:
```suggestion
description: Increase CPU utilisation on all the cores for 60 seconds
```

File: docs/testplan/sample_test_plan.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+  - type: ping_disable
+    config:
+      duration: 60
+      start_delay: 0
+```
+
+#### Block Ports 
+```yaml
+...
+description: Block inbound and outbound connections on port 80 and 8080 for 60 second
```
Comment (r-r-2) [inline docs/testplan/sample_test_plan.md:36]:
```suggestion
description: Block inbound and outbound connections on port 80 and 8080 for 60 seconds
```

File: docs/testplan/sample_test_plan.md [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+        - "https://api.screwdriver.cd"
+        - "https://api.nationalize.io"
+```
+
+#### Endpoint Certificate Validation
+```yaml
+...
+description: Verify endpoint cert expire date
+agents:
+  - type: urls
```
Comment (r-r-2) [inline docs/testplan/sample_test_plan.md:97]: should be server_cert_validation ?
```suggestion
  - type: server_cert_validation
```

File: docs/testplan/sample_test_plan.md [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+        - "www.techcrunch.com"
+        - "www.aol.com"
+```
+!!! note ""
+    For more information on attack agents [refer](/ychaos/agents/)
+
+#### HTTP Request Verification Plugin
+```yaml
+...
+description: Verify Steady state by checking Request latency is bellow 100ms
```
Comment (r-r-2) [inline docs/testplan/sample_test_plan.md:110]:
```suggestion
description: Verify Steady state by checking Request latency is below 100ms
```

File: docs/testplan/sample_test_plan.md [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+attack:
+    target_type: machine
+    agents:
+        - type: no_op
+```
+
+#### Screwdriver V4 Verification Plugin
+```yaml
+...
+description: Use Screwdrive job to verify Steady state
```
Comment (r-r-2) [inline docs/testplan/sample_test_plan.md:134]:
```suggestion
description: Use Screwdriver job to verify Steady state   
```

File: docs/testplan/sample_test_plan.md [hunk 13]
```diff
@@ -0,0 +1,153 @@
+This section of the document contains some example testplans for you to get started quickly on the project
```
Comment (r-r-2) [inline docs/testplan/sample_test_plan.md:1]: if you get time, can you also do a quick validation of the test plans here using cli? ignore if already done.

**Review by r-r-2 — APPROVED** [review]

**Review by shashankrnr32 — APPROVED** [review]

**Pull request merged**